Compiled from "Exception.scala"
public class scala.util.control.Exception$Try extends java.lang.Object implements scala.ScalaObject
  SourceFile: "Exception.scala"
  Signature: length = 0x2
   00 72 
  Scala: length = 0x
   
  InnerClass: 
   public #119= #15 of #118; //Try=class scala/util/control/Exception$Try of class scala/util/control/Exception
   public #120= #74 of #118; //Catch=class scala/util/control/Exception$Catch of class scala/util/control/Exception
   public abstract #121= #50 of #118; //Described=class scala/util/control/Exception$Described of class scala/util/control/Exception
  minor version: 0
  major version: 49
  Constant pool:
const #1 = Asciz	SourceFile;
const #2 = Asciz	Exception.scala;
const #3 = Asciz	catcher;
const #4 = Asciz	Lscala/util/control/Exception$Catch;;
const #5 = Asciz	Lscala/util/control/Exception$Catch<TT;>;;
const #6 = Asciz	Signature;
const #7 = Asciz	body;
const #8 = Asciz	Lscala/Function0;;
const #9 = Asciz	Lscala/Function0<TT;>;;
const #10 = Asciz	<init>;
const #11 = Asciz	(Lscala/Function0;Lscala/util/control/Exception$Catch;)V;
const #12 = Asciz	Code;
const #13 = Asciz	LineNumberTable;
const #14 = Asciz	scala/util/control/Exception$Try;
const #15 = class	#14;	//  scala/util/control/Exception$Try
const #16 = NameAndType	#7:#8;//  body:Lscala/Function0;
const #17 = Field	#15.#16;	//  scala/util/control/Exception$Try.body:Lscala/Function0;
const #18 = NameAndType	#3:#4;//  catcher:Lscala/util/control/Exception$Catch;
const #19 = Field	#15.#18;	//  scala/util/control/Exception$Try.catcher:Lscala/util/control/Exception$Catch;
const #20 = Asciz	java/lang/Object;
const #21 = class	#20;	//  java/lang/Object
const #22 = Asciz	()V;
const #23 = NameAndType	#10:#22;//  "<init>":()V
const #24 = Method	#21.#23;	//  java/lang/Object."<init>":()V
const #25 = Asciz	this;
const #26 = Asciz	Lscala/util/control/Exception$Try;;
const #27 = Asciz	LocalVariableTable;
const #28 = Asciz	(Lscala/Function0<TT;>;Lscala/util/control/Exception$Catch<TT;>;)V;
const #29 = Asciz	toString;
const #30 = Asciz	()Ljava/lang/String;;
const #31 = Asciz	scala/collection/immutable/List$;
const #32 = class	#31;	//  scala/collection/immutable/List$
const #33 = Asciz	MODULE$;
const #34 = Asciz	Lscala/collection/immutable/List$;;
const #35 = NameAndType	#33:#34;//  MODULE$:Lscala/collection/immutable/List$;
const #36 = Field	#32.#35;	//  scala/collection/immutable/List$.MODULE$:Lscala/collection/immutable/List$;
const #37 = Asciz	scala/Predef$;
const #38 = class	#37;	//  scala/Predef$
const #39 = Asciz	Lscala/Predef$;;
const #40 = NameAndType	#33:#39;//  MODULE$:Lscala/Predef$;
const #41 = Field	#38.#40;	//  scala/Predef$.MODULE$:Lscala/Predef$;
const #42 = Asciz	java/lang/String;
const #43 = class	#42;	//  java/lang/String
const #44 = Asciz	Try(<body>);
const #45 = String	#44;	//  Try(<body>)
const #46 = Asciz	()Lscala/util/control/Exception$Catch;;
const #47 = NameAndType	#3:#46;//  catcher:()Lscala/util/control/Exception$Catch;
const #48 = Method	#15.#47;	//  scala/util/control/Exception$Try.catcher:()Lscala/util/control/Exception$Catch;
const #49 = Asciz	scala/util/control/Exception$Described;
const #50 = class	#49;	//  scala/util/control/Exception$Described
const #51 = NameAndType	#29:#30;//  toString:()Ljava/lang/String;
const #52 = InterfaceMethod	#50.#51;	//  scala/util/control/Exception$Described.toString:()Ljava/lang/String;
const #53 = Asciz	[Ljava/lang/Object;;
const #54 = class	#53;	//  "[Ljava/lang/Object;"
const #55 = Asciz	wrapRefArray;
const #56 = Asciz	([Ljava/lang/Object;)Lscala/collection/mutable/WrappedArray;;
const #57 = NameAndType	#55:#56;//  wrapRefArray:([Ljava/lang/Object;)Lscala/collection/mutable/WrappedArray;
const #58 = Method	#38.#57;	//  scala/Predef$.wrapRefArray:([Ljava/lang/Object;)Lscala/collection/mutable/WrappedArray;
const #59 = Asciz	apply;
const #60 = Asciz	(Lscala/collection/Seq;)Lscala/collection/immutable/List;;
const #61 = NameAndType	#59:#60;//  apply:(Lscala/collection/Seq;)Lscala/collection/immutable/List;
const #62 = Method	#32.#61;	//  scala/collection/immutable/List$.apply:(Lscala/collection/Seq;)Lscala/collection/immutable/List;
const #63 = Asciz	 ;
const #64 = String	#63;	//   
const #65 = Asciz	scala/collection/TraversableLike;
const #66 = class	#65;	//  scala/collection/TraversableLike
const #67 = Asciz	mkString;
const #68 = Asciz	(Ljava/lang/String;)Ljava/lang/String;;
const #69 = NameAndType	#67:#68;//  mkString:(Ljava/lang/String;)Ljava/lang/String;
const #70 = InterfaceMethod	#66.#69;	//  scala/collection/TraversableLike.mkString:(Ljava/lang/String;)Ljava/lang/String;
const #71 = Asciz	andFinally;
const #72 = Asciz	(Lscala/Function0;)Lscala/util/control/Exception$Try;;
const #73 = Asciz	scala/util/control/Exception$Catch;
const #74 = class	#73;	//  scala/util/control/Exception$Catch
const #75 = Asciz	(Lscala/Function0;)Lscala/util/control/Exception$Catch;;
const #76 = NameAndType	#71:#75;//  andFinally:(Lscala/Function0;)Lscala/util/control/Exception$Catch;
const #77 = Method	#74.#76;	//  scala/util/control/Exception$Catch.andFinally:(Lscala/Function0;)Lscala/util/control/Exception$Catch;
const #78 = NameAndType	#10:#11;//  "<init>":(Lscala/Function0;Lscala/util/control/Exception$Catch;)V
const #79 = Method	#15.#78;	//  scala/util/control/Exception$Try."<init>":(Lscala/Function0;Lscala/util/control/Exception$Catch;)V
const #80 = Asciz	fin;
const #81 = Asciz	(Lscala/Function0<Ljava/lang/Object;>;)Lscala/util/control/Exception$Try<TT;>;;
const #82 = Asciz	or;
const #83 = Asciz	(Lscala/PartialFunction;)Lscala/util/control/Exception$Try;;
const #84 = Asciz	(Lscala/PartialFunction;)Lscala/util/control/Exception$Catch;;
const #85 = NameAndType	#82:#84;//  or:(Lscala/PartialFunction;)Lscala/util/control/Exception$Catch;
const #86 = Method	#74.#85;	//  scala/util/control/Exception$Catch.or:(Lscala/PartialFunction;)Lscala/util/control/Exception$Catch;
const #87 = Asciz	pf;
const #88 = Asciz	Lscala/PartialFunction;;
const #89 = Asciz	<U:Ljava/lang/Object;>(Lscala/PartialFunction<Ljava/lang/Throwable;TU;>;)Lscala/util/control/Exception$Try<TU;>;;
const #90 = Asciz	tryInstead;
const #91 = Asciz	other;
const #92 = Asciz	<U:Ljava/lang/Object;>(Lscala/Function0<TU;>;)Lscala/util/control/Exception$Try<TU;>;;
const #93 = Asciz	either;
const #94 = Asciz	(Lscala/Function0;)Lscala/Either;;
const #95 = NameAndType	#93:#94;//  either:(Lscala/Function0;)Lscala/Either;
const #96 = Method	#74.#95;	//  scala/util/control/Exception$Catch.either:(Lscala/Function0;)Lscala/Either;
const #97 = Asciz	<U:Ljava/lang/Object;>(Lscala/Function0<TU;>;)Lscala/Either<Ljava/lang/Throwable;TU;>;;
const #98 = Asciz	()Lscala/Either;;
const #99 = Asciz	()Lscala/Either<Ljava/lang/Throwable;TT;>;;
const #100 = Asciz	opt;
const #101 = Asciz	(Lscala/Function0;)Lscala/Option;;
const #102 = NameAndType	#100:#101;//  opt:(Lscala/Function0;)Lscala/Option;
const #103 = Method	#74.#102;	//  scala/util/control/Exception$Catch.opt:(Lscala/Function0;)Lscala/Option;
const #104 = Asciz	<U:Ljava/lang/Object;>(Lscala/Function0<TU;>;)Lscala/Option<TU;>;;
const #105 = Asciz	()Lscala/Option;;
const #106 = Asciz	()Lscala/Option<TT;>;;
const #107 = Asciz	(Lscala/Function0;)Ljava/lang/Object;;
const #108 = NameAndType	#59:#107;//  apply:(Lscala/Function0;)Ljava/lang/Object;
const #109 = Method	#74.#108;	//  scala/util/control/Exception$Catch.apply:(Lscala/Function0;)Ljava/lang/Object;
const #110 = Asciz	<U:Ljava/lang/Object;>(Lscala/Function0<TU;>;)TU;;
const #111 = Asciz	()Ljava/lang/Object;;
const #112 = Asciz	()TT;;
const #113 = Asciz	()Lscala/util/control/Exception$Catch<TT;>;;
const #114 = Asciz	<T:Ljava/lang/Object;>Ljava/lang/Object;Lscala/ScalaObject;;
const #115 = Asciz	Scala;
const #116 = Asciz	InnerClasses;
const #117 = Asciz	scala/util/control/Exception;
const #118 = class	#117;	//  scala/util/control/Exception
const #119 = Asciz	Try;
const #120 = Asciz	Catch;
const #121 = Asciz	Described;
const #122 = Asciz	scala/ScalaObject;
const #123 = class	#122;	//  scala/ScalaObject

{
public scala.util.control.Exception$Try(scala.Function0, scala.util.control.Exception$Catch);
  Code:
   Stack=2, Locals=3, Args_size=3
   0:	aload_0
   1:	aload_1
   2:	putfield	#17; //Field body:Lscala/Function0;
   5:	aload_0
   6:	aload_2
   7:	putfield	#19; //Field catcher:Lscala/util/control/Exception$Catch;
   10:	aload_0
   11:	invokespecial	#24; //Method java/lang/Object."<init>":()V
   14:	return
  LineNumberTable: 
   line 117: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      15      0    this       Lscala/util/control/Exception$Try;
   0      15      1    body       Lscala/Function0;
   0      15      2    catcher       Lscala/util/control/Exception$Catch;

  Signature: length = 0x2
   00 1C 

public java.lang.String toString();
  Code:
   Stack=6, Locals=1, Args_size=1
   0:	getstatic	#36; //Field scala/collection/immutable/List$.MODULE$:Lscala/collection/immutable/List$;
   3:	getstatic	#41; //Field scala/Predef$.MODULE$:Lscala/Predef$;
   6:	iconst_2
   7:	anewarray	#43; //class java/lang/String
   10:	dup
   11:	iconst_0
   12:	ldc	#45; //String Try(<body>)
   14:	aastore
   15:	dup
   16:	iconst_1
   17:	aload_0
   18:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   21:	invokeinterface	#52,  1; //InterfaceMethod scala/util/control/Exception$Described.toString:()Ljava/lang/String;
   26:	aastore
   27:	checkcast	#54; //class "[Ljava/lang/Object;"
   30:	invokevirtual	#58; //Method scala/Predef$.wrapRefArray:([Ljava/lang/Object;)Lscala/collection/mutable/WrappedArray;
   33:	invokevirtual	#62; //Method scala/collection/immutable/List$.apply:(Lscala/collection/Seq;)Lscala/collection/immutable/List;
   36:	ldc	#64; //String  
   38:	invokeinterface	#70,  2; //InterfaceMethod scala/collection/TraversableLike.mkString:(Ljava/lang/String;)Ljava/lang/String;
   43:	areturn
  LineNumberTable: 
   line 139: 0


public scala.util.control.Exception$Try andFinally(scala.Function0);
  Code:
   Stack=5, Locals=2, Args_size=2
   0:	new	#15; //class scala/util/control/Exception$Try
   3:	dup
   4:	aload_0
   5:	getfield	#17; //Field body:Lscala/Function0;
   8:	aload_0
   9:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   12:	aload_1
   13:	invokevirtual	#77; //Method scala/util/control/Exception$Catch.andFinally:(Lscala/Function0;)Lscala/util/control/Exception$Catch;
   16:	invokespecial	#79; //Method "<init>":(Lscala/Function0;Lscala/util/control/Exception$Catch;)V
   19:	areturn
  LineNumberTable: 
   line 137: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      20      0    this       Lscala/util/control/Exception$Try;
   0      20      1    fin       Lscala/Function0;

  Signature: length = 0x2
   00 51 

public scala.util.control.Exception$Try or(scala.PartialFunction);
  Code:
   Stack=5, Locals=2, Args_size=2
   0:	new	#15; //class scala/util/control/Exception$Try
   3:	dup
   4:	aload_0
   5:	getfield	#17; //Field body:Lscala/Function0;
   8:	aload_0
   9:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   12:	aload_1
   13:	invokevirtual	#86; //Method scala/util/control/Exception$Catch.or:(Lscala/PartialFunction;)Lscala/util/control/Exception$Catch;
   16:	invokespecial	#79; //Method "<init>":(Lscala/Function0;Lscala/util/control/Exception$Catch;)V
   19:	areturn
  LineNumberTable: 
   line 134: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      20      0    this       Lscala/util/control/Exception$Try;
   0      20      1    pf       Lscala/PartialFunction;

  Signature: length = 0x2
   00 59 

public scala.util.control.Exception$Try tryInstead(scala.Function0);
  Code:
   Stack=4, Locals=2, Args_size=2
   0:	new	#15; //class scala/util/control/Exception$Try
   3:	dup
   4:	aload_1
   5:	aload_0
   6:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   9:	invokespecial	#79; //Method "<init>":(Lscala/Function0;Lscala/util/control/Exception$Catch;)V
   12:	areturn
  LineNumberTable: 
   line 131: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      13      0    this       Lscala/util/control/Exception$Try;
   0      13      1    other       Lscala/Function0;

  Signature: length = 0x2
   00 5C 

public scala.Either either(scala.Function0);
  Code:
   Stack=2, Locals=2, Args_size=2
   0:	aload_0
   1:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   4:	aload_1
   5:	invokevirtual	#96; //Method scala/util/control/Exception$Catch.either:(Lscala/Function0;)Lscala/Either;
   8:	areturn
  LineNumberTable: 
   line 128: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      9      0    this       Lscala/util/control/Exception$Try;
   0      9      1    other       Lscala/Function0;

  Signature: length = 0x2
   00 61 

public scala.Either either();
  Code:
   Stack=2, Locals=1, Args_size=1
   0:	aload_0
   1:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   4:	aload_0
   5:	getfield	#17; //Field body:Lscala/Function0;
   8:	invokevirtual	#96; //Method scala/util/control/Exception$Catch.either:(Lscala/Function0;)Lscala/Either;
   11:	areturn
  LineNumberTable: 
   line 127: 0

  Signature: length = 0x2
   00 63 

public scala.Option opt(scala.Function0);
  Code:
   Stack=2, Locals=2, Args_size=2
   0:	aload_0
   1:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   4:	aload_1
   5:	invokevirtual	#103; //Method scala/util/control/Exception$Catch.opt:(Lscala/Function0;)Lscala/Option;
   8:	areturn
  LineNumberTable: 
   line 124: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      9      0    this       Lscala/util/control/Exception$Try;
   0      9      1    other       Lscala/Function0;

  Signature: length = 0x2
   00 68 

public scala.Option opt();
  Code:
   Stack=2, Locals=1, Args_size=1
   0:	aload_0
   1:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   4:	aload_0
   5:	getfield	#17; //Field body:Lscala/Function0;
   8:	invokevirtual	#103; //Method scala/util/control/Exception$Catch.opt:(Lscala/Function0;)Lscala/Option;
   11:	areturn
  LineNumberTable: 
   line 123: 0

  Signature: length = 0x2
   00 6A 

public java.lang.Object apply(scala.Function0);
  Code:
   Stack=2, Locals=2, Args_size=2
   0:	aload_0
   1:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   4:	aload_1
   5:	invokevirtual	#109; //Method scala/util/control/Exception$Catch.apply:(Lscala/Function0;)Ljava/lang/Object;
   8:	areturn
  LineNumberTable: 
   line 120: 0

  LocalVariableTable: 
   Start  Length  Slot  Name   Signature
   0      9      0    this       Lscala/util/control/Exception$Try;
   0      9      1    other       Lscala/Function0;

  Signature: length = 0x2
   00 6E 

public java.lang.Object apply();
  Code:
   Stack=2, Locals=1, Args_size=1
   0:	aload_0
   1:	invokevirtual	#48; //Method catcher:()Lscala/util/control/Exception$Catch;
   4:	aload_0
   5:	getfield	#17; //Field body:Lscala/Function0;
   8:	invokevirtual	#109; //Method scala/util/control/Exception$Catch.apply:(Lscala/Function0;)Ljava/lang/Object;
   11:	areturn
  LineNumberTable: 
   line 119: 0

  Signature: length = 0x2
   00 70 

public scala.util.control.Exception$Catch catcher();
  Code:
   Stack=1, Locals=1, Args_size=1
   0:	aload_0
   1:	getfield	#19; //Field catcher:Lscala/util/control/Exception$Catch;
   4:	areturn
  LineNumberTable: 
   line 117: 0

  Signature: length = 0x2
   00 71 

}

